db/migrate/20210316092758_deduplicate_canonical_profiles.rb

```diff
@@ -0,0 +1,5 @@
+class DeduplicateCanonicalProfiles < ActiveRecord::Migration[5.2]
+  def up
+    DuplicateCanonicalProfileResolver.run!
+  end
```

Comment:
In the same migration (same transaction), we want to add a partial index that enforces canonical profile uniqueness. If it's in a different migration, there is some small chance that more duplicates are inserted before our index is created.

Reply:
We can only do this if we fail the migration if there are child profiles under a duplicate one :thinking:

Reply:
Yes, though the index will fail to be created if that happens, and the transaction will be rolled back, so I don't think we need to add any special code to handle that case.

Reply:
Yeah, but would not that break the migration chain and not allow us to write any other migration afterwards? Or is that the goal? :see_no_evil:

Reply:
Yes, it will break the migrations and prevent us from deploying. If this migration fails, we'll have to revisit `DuplicateCanonicalProfileResolver` to reassign children profiles. Optionally, we could go ahead and handle that case now, though it's _probably_ unnecessary (up to you). We could also run the SQL query against prod again before we deploy this to ensure no children profiles exist. @vkrizan any input about how you think we should proceed?

Reply:
If the whole migration chain isn't rollbacked on any error within the chain, then having it within one migration step/file is our best option.